code/weapon/weapons.cpp

```diff
@@ -4395,7 +4408,8 @@ void weapon_home(object *obj, int num, float frame_time)
             (old_dot < wip->fov) &&
             (wp->lifeleft > 0.01f) &&
             (wp->homing_object != &obj_used_list) &&
-            (wp->homing_object->type == OBJ_SHIP))
+            (wp->homing_object->type == OBJ_SHIP) && 
+            (wp->missile_list_index != -1))				// Asteroth - that is, homing primaries don't do this
```

Comment:
I think it would be clearer to check for `WP_LASER` (or `WP_MISSILE`) here because that's conceptually closer to how you are making the distinction.